pkg/webhook/parsers.go

```diff
@@ -27,27 +27,34 @@ import (
 
 var minimumSupportedVersion = version.MustParseGeneric("1.29.0")
 
-// iterates the container list,
-// if a container is named "gke-gcsfuse-sidecar",
-// extract the container image and check if the image is valid,
-// then removes this container from the container list.
+// parseSidecarContainerImage supports our Privately Hosted Sidecar Image option
+// by iterating the container list and finding a container named "gke-gcsfuse-sidecar"
+// If we find "gke-gcsfuse-sidecar":
+//   - extract the container image and check if the image is valid
+//   - removes the container definition from the container list.
+//   - remove any mentions of "gke-gcsfuse-sidecar" from initContainer list.
+//   - return image
 func parseSidecarContainerImage(pod *corev1.Pod) (string, error) {
 	var image string
-	var index int
-	for i, c := range pod.Spec.Containers {
-		if c.Name == SidecarContainerName {
-			image = c.Image
-			index = i
-
-			if _, _, _, err := parsers.ParseImageName(image); err != nil {
-				return "", fmt.Errorf("could not parse input image: %q, error: %w", image, err)
-			}
+
+	// Find container named "gke-gcsfuse-sidecar" (SidecarContainerName), extract its image, and remove from list.
+	if index, present := containerPresent(pod.Spec.Containers, SidecarContainerName); present {
+		image = pod.Spec.Containers[index].Image
+
+		if _, _, _, err := parsers.ParseImageName(image); err != nil {
+			return "", fmt.Errorf("could not parse input image: %q, error: %w", image, err)
```

Comment:
Although we are returning an error, we should not block pod creation because we could run into the situation where the sidecar . Maybe if we encounter this we can use the default sidecar container image contained in the ConfigMap.

Reply:
What is the situation?

Reply:
Resolving as we can keep as is.

This is related to the case where we provide a private sidecar image sidecar where the image is not valid. I think its okay to return the error.